### PR TITLE
Svdesu/ch74/select custom circuit icon

### PR DESCRIPTION
**Pull request by SVD6**
Known Issues: Size of the icon when displayed back in the create a circuit page doesn't match the design layout, but I think I'm gonna change the layout to work around the image as opposed to the other way around. (Easier this way).

What I've done:
- Popup dialog when you click the little icon thing next to circuit name edittext
- Popup has the nice looking carousel picker for icons
- You can select an image and it will change the icon next to the circuit name
- The image is then saved under the circuit object to be displayed wherever else it is needed

### CODE REVIEW

File: app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt [hunk 1]
```diff
@@ -37,6 +43,67 @@ class CircuitCreate : BaseActivity() {
...
+                }
+            })
+
+            // Button Logic
+            dialogView.negative_button.setOnClickListener {
+                builder.dismiss()
+            }
+
+            dialogView.positive_button.setOnClickListener {
+                setStartIcon(selectedIcon)
```
Comment (harigovind1998) [inline app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt]: Add the timer icon at the the beginning and set it as default.

File: app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt [hunk 2]
```diff
@@ -37,6 +43,67 @@ class CircuitCreate : BaseActivity() {
             }
         }
 
+        bind!!.nameInput.setStartIconOnClickListener {
+            val builder = MaterialAlertDialogBuilder(this).create()
+            val dialogView = layoutInflater.inflate(R.layout.dialog_circuit_icon, null)
+
+            // Setting Carousel Items
+            val imageItems: ArrayList<CarouselPicker.PickerItem> = ArrayList()
```
Comment (harigovind1998) [inline app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt]: Move this out into its own function. Will be cleaner to read and modify.

File: app/src/main/res/drawable/ic_abs.xml [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
+    android:width="50dp"
+    android:height="50dp"
+    android:viewportWidth="512"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_abs.xml:4]: I Think you ideally want the width/height and viewport dimensions to the same. https://stackoverflow.com/questions/41353732/what-is-the-impact-of-viewportwidth-height-on-android-drawables

File: app/src/main/res/drawable/ic_gym_3.xml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="469.333dp"
-    android:height="469.333dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_gym_3.xml:2]: same here

File: app/src/main/res/drawable/ic_boxer.xml [hunk 5]
```diff
@@ -1,9 +1,9 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="512dp"
-    android:height="512dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_boxer.xml:2]: here

File: app/src/main/res/drawable/ic_dumbbell.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
+    android:height="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_dumbbell.xml:3]: here

File: app/src/main/res/drawable/ic_gym.xml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="512dp"
-    android:height="512dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_gym.xml:2]: here

File: app/src/main/res/drawable/ic_gym_2.xml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="512dp"
-    android:height="512dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_gym_2.xml:2]: here

File: app/src/main/res/drawable/ic_gym_4.xml [hunk 9]
```diff
@@ -1,6 +1,6 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="512.009dp"
-    android:height="512.009dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_gym_4.xml:2]: here

File: app/src/main/res/drawable/ic_gym_bag.xml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_gym_bag.xml:2]: here

File: app/src/main/res/drawable/ic_gymnast.xml [hunk 11]
```diff
@@ -1,12 +1,12 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="512dp"
-    android:height="512dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_gymnast.xml:2]: here

File: app/src/main/res/drawable/ic_jump_rope.xml [hunk 12]
```diff
@@ -1,9 +1,9 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="512dp"
-    android:height="512dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_jump_rope.xml:2]: here

File: app/src/main/res/drawable/ic_mat.xml [hunk 13]
```diff
@@ -1,15 +1,15 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="512.034dp"
-    android:height="512.034dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_mat.xml:2]: here

File: app/src/main/res/drawable/ic_punching_ball.xml [hunk 14]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_punching_ball.xml:2]: here

File: app/src/main/res/drawable/ic_resistance.xml [hunk 15]
```diff
@@ -0,0 +1,15 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_resistance.xml:2]: here

File: app/src/main/res/drawable/ic_resistance_1.xml [hunk 16]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_resistance_1.xml:2]: here

File: app/src/main/res/drawable/ic_treadmill.xml [hunk 17]
```diff
@@ -1,12 +1,12 @@
 <vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="512dp"
-    android:height="512dp"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_treadmill.xml:2]: here

File: app/src/main/res/drawable/ic_workout.xml [hunk 18]
```diff
@@ -0,0 +1,12 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
+    android:height="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_workout.xml:3]: here

File: app/src/main/res/drawable/ic_workout_3.xml [hunk 19]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_workout_3.xml:2]: here

File: app/src/main/java/com/example/chrono/util/UtilClass.kt [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+        11 -> return "ic_jump_rope"
+        12 -> return "ic_mat"
+        13 -> return "ic_punching_ball"
+        14 -> return "ic_resistance"
+        15 -> return "ic_resistance_1"
+        16 -> return "ic_treadmill"
+        17 -> return "ic_workout"
+        18 -> return "ic_workout_3"
+    }
+    return "-1"
```
Comment (harigovind1998) [inline app/src/main/java/com/example/chrono/util/UtilClass.kt:32]: Throw an Exception

File: app/src/main/res/drawable/ic_arm.xml [hunk 21]
```diff
@@ -1,9 +1,9 @@
...
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_arm.xml:2]: here

File: app/src/main/res/drawable/ic_bottle.xml [hunk 22]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
+    android:height="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_bottle.xml:3]: here

File: app/src/main/res/drawable/ic_boxer.xml (hunk 5 above)
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_boxer.xml:2]: here

File: app/src/main/res/drawable/ic_dumbbell.xml [hunk 23]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="50dp"
```
Comment (harigovind1998) [inline app/src/main/res/drawable/ic_dumbbell.xml:2]: here

File: app/src/main/res/layout/dialog_circuit_icon.xml [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+
+                <LinearLayout
+                    android:id="@+id/content"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_below="@+id/title"
+                    android:orientation="vertical">
+
+                    <TextView
+                        android:id="@+id/icon_name"
```
Comment (harigovind1998) [inline app/src/main/res/layout/dialog_circuit_icon.xml]: This is unused I believe.

File: app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt [hunk 25]
```diff
@@ -57,12 +124,18 @@ class CircuitCreate : BaseActivity() {
         }
     }
 
+    private fun setStartIcon(selectedIcon: String) {
```
Comment (harigovind1998) [inline app/src/main/java/com/example/chrono/main/circuit/CircuitCreate.kt]: setSelectedIcon() instead of setStartIcon()

**Comment by SVD6** [comment]
Had to use a different icon because the one I had wasn't behaving the same as the others. Also created a different temporary solution for the icon/editext layout. We will discuss this on call with Aarthi to decide how to make it look better.

**Pull request merged**